tests/mock/agents.py

```diff
@@ -1,221 +1,153 @@
-        assert kwargs == agent_kwargs, 'Unexpected agent input'
-
-    def assert_contract_calls(self, calls: Tuple[str]) -> None:
-        for call_name in calls:
-            assert call_name in self._SPY_CALLS, f'"{call_name}" was not called'
-
-
-class MockNucypherToken(MockContractAgent, NucypherTokenAgent):
-    """Look at me im a token!"""
-
-
-class MockStakingAgent(MockContractAgent, StakingEscrowAgent):
-    """dont forget the eggs!"""
-
-    CALLS = ('get_completed_work', )
-
-
-class MockPolicyManagerAgent(MockContractAgent, PolicyManagerAgent):
-    """The best ethereum policy manager ever"""
-
-
-class MockWorkLockAgent(MockContractAgent, WorkLockAgent):
-
-    CALLS = ('check_claim',
-             'eth_to_tokens',
-             'get_deposited_eth',
-             'get_eth_supply',
-             'get_base_deposit_rate',
-             'get_bonus_lot_value',
-             'get_bonus_deposit_rate',
-             'get_bonus_refund_rate',
-             'get_base_refund_rate',
-             'get_completed_work',
-             'get_refunded_work')
-
-    TRANSACTIONS = ('bid',
-                    'cancel_bid',
-                    'force_refund',
-                    'verify_bidding_correctness',
-                    'claim',
-                    'refund',
-                    'withdraw_compensation')
-
-    def __init__(self):
-
-        # Allow for mocking
-        economics = EconomicsFactory.get_economics(registry=Mock())
-
-        self.ATTRS = {'boosting_refund': economics.worklock_boosting_refund_rate,
-                      'slowing_refund': 1,  # TODO: another way to get this value?
-                      'start_bidding_date': economics.bidding_start_date,
-                      'end_bidding_date': economics.bidding_end_date,
-                      'end_cancellation_date': economics.cancellation_end_date,
-                      'minimum_allowed_bid': economics.worklock_min_allowed_bid,
-                      'lot_value': economics.worklock_supply}
+    @property
+    def all_transactions(self) -> List[Callable]:
+        interface = ContractInterfaces.TRANSACTION
+        transaction_functions = self.__get_interface_calls(interface=interface)
+        return transaction_functions
+
+    @property
+    def contract_calls(self) -> List[Callable]:
+        interface = ContractInterfaces.CALL
+        transaction_functions = self.__get_interface_calls(interface=interface)
+        return transaction_functions
+
+    def get_unexpected_transactions(self, allowed: Union[Iterable[Callable], None]) -> List[Callable]:
+        if allowed:
+            predicate = lambda tx: tx not in allowed and tx.called
+        else:
+            predicate = lambda tx: tx.called
+        unexpected_transactions = list(filter(predicate, self.all_transactions))
+        return unexpected_transactions
 
-        super().__init__()
+    def assert_no_unexpected_transactions(self, allowed: Iterable[Callable]) -> None:
```

Comment:
double negation in name reads difficult, at least for me

Reply:
Yeah naming is tricky here.  I want to assert that there are no unintentional side effects in the form of a transaction. ...Suggestions?

Reply:
`assert_only_expected_transactions`? I like when something starts `assert_only`